### PR TITLE
Note that dhparam does support X9.42

### PR DESCRIPTION
**Pull request by richsalz**
Fixes: #13151

### CODE REVIEW

**Review by slontis — COMMENTED** [review]
This interface is legacy as far as I am concerned.. There are 4 type of DH keygen (this only does 2 of them).
Using genpkey with a named group should be the recommended way for DH.

**Comment by richsalz** [comment]
> This interface is legacy as far as I am concerned.. There are 4 type of DH keygen (this only does 2 of them).
> Using genpkey with a named group should be the recommended way for DH.

Thoughts for what it should say?  It would be funny but adding "This command itself is superseded by genpkey (perhaps with a named groups) and XXX" right after the "replaces" sentence. :)  Happy to do that, XXX is what?

**Comment by slontis** [comment]
Something like:

See L<openssl-genpkey(1)/EXAMPLES> to Generate a DH key using a named safe prime group without generating intermediate parameters.

The FIPS186-4 code is really a deprecated usecase - so probably not even worth mentioning.

**Comment by richsalz** [comment]
Thanks @slontis, made some changes and pushed an updated commit.

File: doc/man1/openssl-dhparam.pod.in [hunk 1]
```diff
@@ -28,7 +28,11 @@ B<openssl dhparam>
 
 =head1 DESCRIPTION
 
-This command is used to manipulate DH parameter files.
+This command is used to manipulate DH and DSA parameter files.
```
Comment (slontis) [inline doc/man1/openssl-dhparam.pod.in:31]: Maybe leave the DSA bit out? The main purpose of this app is to generate DH params?

Comment (richsalz) [reply doc/man1/openssl-dhparam.pod.in:31]: okay I will leave out the DSA bit.

Comment (richsalz) [reply doc/man1/openssl-dhparam.pod.in]: udpated commit pushed.

**Review by slontis — APPROVED** [review]

**Review by paulidale — APPROVED** [review]

**Comment by paulidale** [comment]
Merged to master, thanks.

**Pull request closed without merging**